src/main/java/com/limechain/beefy/dto/ValidatorSet.java

```diff
@@ -1,12 +1,16 @@
 package com.limechain.beefy.dto;
 
-import io.libp2p.core.crypto.PubKey;
+import lombok.AllArgsConstructor;
 import lombok.Data;
+import lombok.NoArgsConstructor;
 
+import java.math.BigInteger;
 import java.util.List;
 
 @Data
+@AllArgsConstructor
+@NoArgsConstructor
```

Comment:
```suggestion
@Value
```

Value enforces private final fields. As we receive the validator set from a digest and do not change any data afterwards it is a suitable candidate to enforce this annotation.